fml/memory/weak_ptr.h

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file provides weak pointers and weak pointer factories that work like
+// Chromium's |base::WeakPtr<T>| and |base::WeakPtrFactory<T>|.
+
+#ifndef FLUTTER_FML_MEMORY_WEAK_PTR_H_
+#define FLUTTER_FML_MEMORY_WEAK_PTR_H_
+
+#include <utility>
+
+#include "lib/fxl/logging.h"
+#include "lib/fxl/memory/ref_counted.h"
+#include "lib/fxl/memory/weak_ptr_internal.h"
```

Comment:
Would you be willing to make you own copy of the internal details too?  It's not ideal to have an external dependency on the internal details of fxl's weak ptr.

Reply:
On it.